src/components/structures/login/Login.js

```diff
@@ -125,7 +123,7 @@ module.exports = React.createClass({displayName: 'Login',
             // logins so let's skip that for now).
             loginLogic.chooseFlow(0);
         }, function(err) {
-            self._setErrorTextFromError(err);
+            self._setStateFromError(err);
```

Comment:
hrm. if we get a 40x here does it really make sense to set `loginIncorrect=true`?